avcodec.cpp

```diff
@@ -411,7 +416,14 @@ bool avcodec_decoder_has_subtitles(const avcodec_decoder d)
 
 static int avcodec_decoder_copy_frame(const avcodec_decoder d, opencv_mat mat, AVFrame* frame)
 {
+    if (!d || !d->codec || !d->codec->codec || !mat || !frame) {
+        return -1;
```

Comment:
at some point we should probably move to more modern c++ features like `std::optional/std::expected` instead of spitting out `-1` for errs no?

Reply:
For sure - in the meantime I think we should stick with numeric return codes, as C++ exceptions have bitten us recently.